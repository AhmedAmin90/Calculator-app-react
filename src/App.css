.App {
  text-align: center;
}

nav {
  margin: 30px auto;
  width: 75%;
  border-radius: 10px;
  background-color:#455a7e;
  height: 50px;
  display: flex;
  justify-content: space-around;
  align-items: center;
}

nav a {
  color: white;
  text-decoration: none;
  text-transform: capitalize;
  font-size: 20px;
  font-weight: bold;
}

nav a:hover {
 color: #EFD170;;
}