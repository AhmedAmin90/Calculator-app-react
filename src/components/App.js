import React from 'react';
import {
  Switch,
  Route,
  NavLink,
  BrowserRouter as Router
} from 'react-router-dom';
import Calculator from './Calculator';
import About from './About';
import Quote from './Quote';
import '../App.css';

const App = () => (
  <Router >
  <div className="App">
    <nav>

      <NavLink to="/"> Home </NavLink>
      <NavLink to="/calculator"> Calculator </NavLink>
      <NavLink to="/quote"> Quote </NavLink>

    </nav>

    <Switch>
      <Route exact path="/" render={() => <About />} />
      <Route exact path="/calculator" render={() => <Calculator />} />
      <Route exact path="/quote" render={() => <Quote />} />
    </Switch>

    

  </div>
  </Router>
);
export default App;
