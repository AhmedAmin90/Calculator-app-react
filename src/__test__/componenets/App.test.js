import React from 'react';
import { create } from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import App from '../../components/App'


test('renders correctly', () => {
  const tree = create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('displays heading', () => {
  render(<App />);
  expect(screen.getByRole('heading')).toHaveTextContent('AHMED AMIM WEB DEVELOPER');
});
