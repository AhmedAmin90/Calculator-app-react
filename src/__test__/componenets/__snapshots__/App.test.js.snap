exports[`test renders correctly 1`] = `
<div
  className="App">
  <nav>
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
      style={Object {}}>
       Home 
    </a>
    <a
      aria-current={null}
      className={undefined}
      href="/calculator"
      onClick={[Function]}
      style={undefined}>
       Calculator 
    </a>
    <a
      aria-current={null}
      className={undefined}
      href="/quote"
      onClick={[Function]}
      style={undefined}>
       Quote 
    </a>
  </nav>
  <div
    className="About">
    <div
      className="hero-section-text">
      <h1>
        AHMED AMIM
        <br />
         
        WEB DEVELOPER
      </h1>
      <p
        className="my-description">
        I’m a software developer! I can help you build a product , feature or website Look through some of my work and experience! If you like what you see and have a project you need coded, don’t hestiate to contact me.
      </p>
      <p
        className="connect-text">
         lets Connect
      </p>
      <ul
        className="social-icons">
        <li>
           
          <a
            href="https://twitter.com/AhmedAmin12383"
            rel="noreferrer"
            target="_blank">
            <i
              className="fab fa-twitter" />
             
             
          </a>
           
        </li>
        <li>
           
          <a
            href="https://www.linkedin.com/in/web-developer/"
            rel="noreferrer"
            target="_blank">
             
            <i
              className="fab fa-linkedin-in" />
             
          </a>
           
        </li>
        <li>
           
          <a
            href="https://github.com/AhmedAmin90"
            rel="noreferrer"
            target="_blank">
             
            <i
              className="fab fa-github" />
             
          </a>
        </li>
        <li>
           
          <a
            href="https://angel.co/u/ahmed-amin-22"
            rel="noreferrer"
            target="_blank">
             
            <i
              className="fab fa-angellist" />
             
          </a>
        </li>
        <li>
           
          <a
            href="https://www.facebook.com/ahmed.amin.7564"
            rel="noreferrer"
            target="_blank">
             
            <i
              className="fab fa-facebook-f" />
             
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
